deep-sea-stories/packages/backend/src/context.ts

```diff
@@ -11,4 +12,12 @@ export function createContext({ req, res }: CreateFastifyContextOptions) {
 	return { req, res, fishjam };
 }
 
-export type Context = Awaited<ReturnType<typeof createContext>>;
+export function createWSContext({ req, res }: CreateWSSContextFnOptions) {
+	return { req, res, fishjam };
+}
+
+export type Context = {
+	req: CreateFastifyContextOptions['req'] | CreateWSSContextFnOptions['req'];
+	res: CreateFastifyContextOptions['res'] | CreateWSSContextFnOptions['res'];
+	fishjam: FishjamClient;
+};
```

Comment:
Isn't this more accurate?

```suggestion
export type Context = {
	req: CreateFastifyContextOptions['req'];
	res: CreateFastifyContextOptions['res'];
	fishjam: FishjamClient;
} | {
	req: CreateWSSContextFnOptions['req'];
	res: CreateWSSContextFnOptions['res'];
	fishjam: FishjamClient;
};
```